lib/nm/index.ts

```diff
@@ -0,0 +1,121 @@
+import * as Promise from 'bluebird';
+import * as _ from 'lodash';
+import NetworkManagerTypes from './types';
+
+const dbus: any = require('dbus-native');
+
+const systemBus: any = dbus.systemBus();
+const service: string[] = [ 'org', 'freedesktop', 'NetworkManager' ];
+
+export interface WifiDevice {
+	devicePath: string;
+	type: number;
+}
+
+export interface AccessPoint {
+	apPath: string;
+	apSsid: string;
+}
+
+export class NetworkManager extends NetworkManagerTypes {
+	wifiDevice: WifiDevice;
+	accessPoints: AccessPoint[];
+	wifiActive: boolean;
+
+	constructor() {
+		super();
+
+		this.wifiActive = true;
+
+		this.getWifiDevice()
+		.then((device: WifiDevice) => {
+			const service = Promise.promisifyAll(systemBus.getService('org.freedesktop.NetworkManager'));
+			service.getInterfaceAsync(device.devicePath, 'org.freedesktop.NetworkManager.Device.Wireless')
+			.then((iface) => {
+				console.log('YOLO');
+				iface.on('AccessPointAdded', (...data) => {
+					console.log('YAY ACCESS POINT ADDED', ...data);
+				});
+				iface.on('AccessPointRemoved', (...data) => {
+					console.log('YAY ACCESS POINT REMOVED', ...data);
+				});
+				console.log('SBOLO');
+			})
+			.catch((err) => {
+				console.error('gnappo', err);
+			});
+		});
+	}
+
+	toggleWifi(value) {
+		return this.wifiActive = value;
```

Comment:
What is the point of returning `value`?

Reply:
The point is to let the toggler know the new state of the wifi, since this method is gonna return a Promise (should be ready in 10 minutes) and the Promise could also fail.
It's no big deal, just a safety measure